### PR TITLE
Stop using contains for queries which use labels to match

### PR DESCRIPTION
**Pull request by joelbyler**
I'm not sure if this is the perfect fix for #306 but I was able to get the new test to pass using this revised xpath.  That thing is so hard to follow though, I'm sure I might have missed something.  I'm opening this in the hopes to start a conversation about fixing this issue and I'm glad to help out.  If this does actually fix the issue then 🎉, but I'm betting there's more to it.

Also, what are your thoughts on making that xpaths easier to parse visually?  I spent some time breaking bits of it apart locally but didn't commit any of that here.  Perhaps I can offer some help there to lower the bar of entry for others, let me know.  

Thanks for reviewing and thanks in advance for your feedback!

### CODE REVIEW

**Comment by joelbyler** [comment]
... and there is a check that labels have a for attribute.  Hmm...  I'll look into that.

**Comment by keathley** [comment]
@joelbyler Thanks for working on this! I'm sorry about the xpath stuff. It's gross. I originally intended on building a library to compose xpath queries but never made time for it. I haven't come up with a better solution yet but if you have thoughts I'd be interested in hearing them.

**Comment by fusillicode** [comment]
Hi guys 😄 

is this still relevant? I would love to align this PR with the latest master if needed 😊

P.S: thanks a lot for this great library! 🙇
